editoast/src/views/timetable.rs

```diff
@@ -449,21 +458,49 @@ impl Conflict {
 }
 
 #[derive(Debug, Clone, PartialEq, Deserialize, Serialize, ToSchema)]
-struct PacedTrainOccurrenceId {
-    paced_train_id: i64,
-    index: u64,
+#[serde(untagged)]
+enum ConflictPacedTrainId {
+    Occurence {
+        paced_train_id: i64,
+        index: u64,
+    },
+    Exception {
+        paced_train_id: i64,
+        exception_key: String,
+        index: Option<u64>,
+    },
 }
```

Comment:
I have a lot of concerns about the modifications to this type. Below is a proposition that summarize my above comments, what do you think (cc @younesschrifi which might also be interested)?

```rust
#[derive(Debug, Clone, PartialEq, Deserialize, Serialize, ToSchema)]
struct PacedTrainOccurrenceId {
  paced_train_id: i64,
  occurrence_ref: OccurrenceRef,
}

#[derive(Debug, Clone, PartialEq, Deserialize, Serialize, ToSchema)]
enum OccurrenceRef { // I'm opened to suggestion for the name of this struct
  Base { // If I remember, we use the term 'base' to describe an unmodified occurrence
    index: u64,
  },
  Modified {
    index: u64,
    exception_key: String,
  },
  Added {
    exception_key: String,
  },
}
```

Reply:
I agree, I think the enum idea is better. To be honest, we were in favor of the `Option<i64>` at first, but the enum idea makes for greater clarity in the code and avoids the need to manipulate `Option`.